File: Backend/ExpensePilot.API/ExpensePilot.API/Models/Budget.cs
namespace ExpensePilot.API.Models
{
    public class Budget
    {
        public int BudgetId { get; set; }

        public string UserId { get; set; }

        public int CategoryId { get; set; }

        public decimal BudgetAmount { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        // Navigation Properties
        public ApplicationUser? User { get; set; }

        public Category? Category { get; set; }
    }
}

File: AI/app/database/models/budget.py
from sqlalchemy import Column, Integer, String, Numeric, DateTime, ForeignKey

from app.database.base import Base


class Budget(Base):
    __tablename__ = "Budgets"

    budget_id = Column("BudgetId", Integer, primary_key=True, index=True)

    user_id = Column("UserId", String, ForeignKey("AspNetUsers.Id"))

    category_id = Column("CategoryId", Integer, ForeignKey("Categories.CategoryId"))

    budget_amount = Column("BudgetAmount", Numeric(18, 2), nullable=False)

    start_date = Column("StartDate", DateTime, nullable=False)

    end_date = Column("EndDate", DateTime, nullable=False)